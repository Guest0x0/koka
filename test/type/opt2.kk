noinline fun plus(x:int, y:int, z:int=1)
  x+y+z

fun psum( xs:list<int>, ?plus) : int
  xs.foldl(0,plus)

pub fun main()
  [1,2,3].psum(?plus=plus)
