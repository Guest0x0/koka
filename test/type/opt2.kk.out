type/opt2/main: () -> int
type/opt2/plus: (x : int, y : int, z : ? int) -> int
type/opt2/psum: forall<a> (xs : list<int>, ?plus : a) -> int